src/test/java/com/pngencoder/SubimageEncodingTest.java

```diff
@@ -113,10 +116,17 @@ private void validateImage(PngEncoderBufferedImageType type, BufferedImage image
                 long rgbSource = image.getRGB(x, y) & 0xFFFFFFFFL;
                 long rgb1 = img1.getRGB(x, y) & 0xFFFFFFFFL;
                 long rgb2 = img2.getRGB(x, y) & 0xFFFFFFFFL;
-                assertEquals(rgbSource, rgb1, "Source compare failure with type " + type + " "
-                        + Long.toString(rgbSource, 16) + " != " + Long.toString(rgb1, 16));
-                assertEquals(rgb1, rgb2, "Compare failure with type " + type + " " + Long.toString(rgb1, 16) + " != "
-                        + Long.toString(rgb2, 16));
+
+                long a1 = (rgb1 & 0xFF000000) >> 24;
+                long a2 = (rgb1 & 0xFF000000) >> 24;
+
+                // We only compare the image rgb values if the alpha is not 0
+                if (a1 != 0 && a2 != 0) {
```

Comment:
If either is 0, there are no assertions. I'd say _both_ have to be zero, or add an assertion that checks that they have the same alpha.

Reply:
Good catch. This should be an `||` not and `&&` ...

As the rgb values including alpha are compared in the branch an extra assert for alpha is not needed.

Reply:
I'm using `assertThatImageIsEqual` for the compare now, so this loop has been deleted now. See [090813c](https://github.com/pngencoder/pngencoder/pull/33/commits/090813c2150ad92402f52d427d721f1210883bec)

Reply:
Nicely done!